packages/vscode-extension/src/panels/Tabpanel.ts

```diff
@@ -26,7 +26,18 @@ export class TabPanel implements Disposable {
     this._panel.iconPath = Uri.joinPath(context.extensionUri, "assets", "logo.svg");
     // Set an event listener to listen for when the panel is disposed (i.e. when the user closes
     // the panel or when the panel is closed programmatically)
-    this._panel.onDidDispose(() => this.dispose());
+
+    commands.executeCommand("setContext", "RNIDE.isTabPanelFocused", this._panel.active);
+
+    const changeViewEvent = this._panel.onDidChangeViewState((e) => {
+      commands.executeCommand("setContext", "RNIDE.isTabPanelFocused", this._panel.active);
+    });
+
+    this._panel.onDidDispose(() => {
+      changeViewEvent.dispose();
+      commands.executeCommand("setContext", "RNIDE.isTabPanelFocused", false);
```

Comment:
to keep the schema similar to what we have in other classes, I'd prefer to keep all the logic responsible for diposing inside of the `dispose` method from line 97. Can you make it so that `onDidDispose` just calls `this.dispose()` and move the rest of the added code to that method. This will require that we store event listener as class memeber but that's ok as we do the same with other disposable things.

For the cases when there are more than a couple of disposable items we have this `subscriptions` concept that's used in extension.ts but I'd rather not use it here as long as we have just a few things to dispose.

Reply:
@kmagiera thanks for the suggestions I applied them. `This will require that we store event listener as class memeber but that's ok as we do the same with other disposable things.` - yeah.. at first I thought about this and that was the reason I avoided that... however separating things and keeping a consistent approach across code-base makes more sense.